docs/reference/build-environment-options.rst

```diff
@@ -2,3 +2,86 @@
 
 Build environment options
 =========================
+
+Snapcraft can optionally use the following arguments to modify the build environment.
+
+
+.. list-table::
+   :header-rows: 1
+   :widths: 2 1 1
+
+   * - Snapcraft Argument
+     - Description
+     - Notes
+
+   * - ``--destructive-mode``
+     - **Destructive mode**. Designed to be used in scenarios where additional
+       provisioning of the build environment is required. This is not recommended
+       because the build *could* contaminate the host build environment.
+     - See :ref:`destructive mode <reference-build-environment-options-destructive>`.
+
+   * - ``--use-lxd``
+     - Builds the snap using `LXD <https://linuxcontainers.org/lxd/introduction/>`_
+       rather than Multipass. This can potentially reduce resource usage, especially
+       from a VM. You can also set ``SNAPCRAFT_BUILD_ENVIRONMENT=lxd`` to use LXD by
```

Comment:
`snap set snapcraft provider=lxd` also works.

Reply:
If we're going to change this, I think we should link to [Select a build provider](https://canonical-snapcraft.readthedocs-hosted.com/en/stable/howto/select-a-build-provider.html) instead.